### PR TITLE
Add pi unit test

### PR DESCRIPTION
**Pull request by jpdoyle**
Adds a simple unit test to compute pi via a monte-carlo method. Calls srand() with 1+pe_id, so it is deterministic for any fixed PE count.

### CODE REVIEW

File: test/unit/pi.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }
+
+    shmem_barrier_all();
+
+    if(me == 0) {
+        for(int i = 1; i < myshmem_n_pes; ++i) {
+            unsigned long long remoteInside,remoteTotal;
+            shmem_longlong_get(&remoteInside,&inside,1,1);
+            shmem_longlong_get(&remoteTotal,&total,1,1);
```
Comment (jdinan) [inline test/unit/pi.c]: There's a bug in one of the arguments to both get routines.

File: test/unit/pi.c [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            total += remoteTotal;
+            inside += remoteInside;
+        }
+
+        double approx_pi = 4.0*inside/(double)total;
+
+        assert(fabs(M_PI-approx_pi) < 0.1);
```
Comment (jdinan) [inline test/unit/pi.c]: Could you change this from assert to ```if (...) ++errors;``` and return errors below instead of 0?

**Comment by jpdoyle** [comment]
Woops, I'll fix those on Monday.

**Comment by jdinan** [comment]
Great work!  :+1: :shipit:

**Pull request merged**